### PR TITLE
feat: general clean up

### PR DESCRIPTION
**Pull request by bazmatic**
Removed old EC stuff
Used enums
Corrected readme
Reorganised units

### CODE REVIEW

**Review by brettvniekerk — DISMISSED** [review]

**Review by OptimusOpus — DISMISSED** [review]
LGTM, the only thing I would maybe confirm is if we want to use chai for unit testing? I thought we had decided to move back to Jest but I 100% could be wrong as that has changed a few times

**Review by brettvniekerk — DISMISSED** [review]

File: src/services/Example.service.ts [hunk 1]
```diff
@@ -1,17 +1,23 @@
 import { Injectable } from "@nestjs/common";
```
Comment (pondpiu) [inline src/services/Example.service.ts:1]: Build failling due to Capital file name Example.service

Comment (bazmatic) [reply src/services/Example.service.ts:1]: ahh thanks!

**Review by brettvniekerk — APPROVED** [review]

**Review by OptimusOpus — APPROVED** [review]

File: src/database/entities/user.entity.ts [hunk 2]
```diff
@@ -6,19 +6,19 @@ import {
...
     @PrimaryGeneratedColumn("uuid")
-    userId: string;
+    id: string;
 
-    @Column({ unique: true })
+    @Column()
     email: string;
 
-    @Column({ type: "timestamptz" })
+    @Column()
```
Comment (deBhal) [inline src/database/entities/user.entity.ts:17]: No big deal, but I'm curious why you took out the type hints here.

File: src/tokens.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export enum ProviderTokens {
+    UserService = "UserService",
+    ExampleService = "ExampleService",
+    PostgresDataSource = "PostgresDataSource",
+    UserPostgres = "UserPostgres"
```
Comment (deBhal) [inline src/tokens.ts:5]: Oh! this reminds me about trailing commas https://eslint.org/docs/latest/rules/comma-dangle - it's in our spec somewhere and makes for cleaner diffs

File: src/user/user.controller.spec.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { Test, TestingModule } from "@nestjs/testing";
 import { UserController } from "./user.controller";
 import { UserModule } from "./user.module";
 import { expect } from "chai";
-import { DatabaseModule } from "../data/database.module";
+import { DatabaseModule } from "../database/database.module";
```
Comment (deBhal) [inline src/user/user.controller.spec.ts:6]: Also absolute paths - I think the trick with this is that you need to tell jest about it too.

I personally don't have an opinion preference on something like `src/...` vs something like `~/...` `~/src/...` or whatever, but I don't like the way `../` adds an extra chore every time you move a file and a tiny bit of extra mental overhead every time you include one. It's not like either is in any way a problem, but tiny little things add up.

File: src/errors.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+// String values used in user-facing error messages
```
Comment (deBhal) [inline src/errors.ts:1]: As an optional suggestion in this vein, we've quite liked having a `{success: boolean, message?: string}` `GenericResponse` type api endpoints to base return-types on.

File: src/services/example.service.ts [hunk 6]
```diff
@@ -1,17 +1,23 @@
 import { Injectable } from "@nestjs/common";
 import { ConfigService } from "@nestjs/config";
-import { IExampleService } from "../interfaces";
+import { IExampleService } from "./example.types";
 
 /* 
-    This is a service that is not associated with a controller or module.
+    This is a service that is not associated with a controller or module.
```
Comment (deBhal) [inline src/services/example.service.ts:6]: Total nitpick, but it's probably worth being annoying about whitespace in a template

File: src/database/entities/user.entity.ts (hunk 2 above)
Comment (bazmatic) [reply src/database/entities/user.entity.ts:17]: whoops, left over from when I was trying to get Mongo and Postgres working side by side

File: src/tokens.ts (hunk 3 above)
Comment (bazmatic) [reply src/tokens.ts:5]: agreed

File: src/user/user.controller.spec.ts (hunk 4 above)
Comment (bazmatic) [reply src/user/user.controller.spec.ts:6]: I don't think absolute paths can work, can they?

Comment (deBhal) [reply src/user/user.controller.spec.ts:6]: Yeah, there are a few different ways, although I have to admit I started, failed and gave up first time I looked at it.

**Pull request merged**